### DISCUSSION TITLE
PHP error when using stripe

### DISCUSSION THREAD

**Discussion by dietrichmd** [liedekef/events-made-easy#631] [Q&A]
When attempting to save stripe info in settings the site gives a generic Wordpress error, yet the data saves.  On the front end, the user can get all the way to the payment page and when they click pay with stripe, the site gives the generic error.

Upon searching the logs, I see the following:

`[Tue Aug 13 12:22:51.309680 2024] [proxy_fcgi:error] [pid 172621:tid 172621] [client 162.158.174.171:33756] AH01071: Got error 'PHP message: PHP Fatal error:  Cannot declare class Stripe\\Util\\ApiVersion, because the name is already in use in /public_html/wp-content/plugins/events-made-easy/payment_gateways/stripe/lib/Util/ApiVersion.php on line 7', referer: https://domain.com/events/p/13b65855ad97ca23a1d916e0a6f0fb0feda7dea48cfbb8/`

and

`[Tue Aug 13 12:22:30.077070 2024] [proxy_fcgi:error] [pid 173486:tid 173486] [client 162.158.174.171:30458] AH01071: Got error 'PHP message: PHP Warning:  Trying to access array offset on null in /public_html/wp-content/plugins/events-made-easy/eme-functions.php on line 1211; PHP message: PHP Warning:  Trying to access array offset on null in /public_html/wp-content/plugins/events-made-easy/eme-functions.php on line 1035; PHP message: PHP Warning:  Trying to access array offset on null in /public_html/wp-content/plugins/events-made-easy/eme-payments.php on line 379; PHP message: PHP Warning:  Trying to access array offset on null in /public_html/wp-content/plugins/events-made-easy/eme-payments.php on line 3466; PHP message: PHP Warning:  Trying to access array offset on null in /public_html/wp-content/plugins/events-made-easy/eme-payments.php on line 3470', referer: https://domain.com/events/p/e8fdcf52d4b0de7e298567a6faae2e246330638ed72663/`

Generic error is:

```
There has been a critical error on this website.

[Learn more about troubleshooting WordPress.](https://wordpress.org/documentation/article/faq-troubleshooting/)
```

Running PHP 8.3 and WP Latest

Any ideas @liedekef ?

**Comment by liedekef** [comment]
This is because another plugin (or even the theme) also includes (a version of) the stripe library. EME only includes its included version of the stripe library when needed, where the other plugin seems to include their version of the library in/on all pages.
See https://stackoverflow.com/questions/57571172/php-fatal-error-cannot-declare-class-stripe-stripe-because-the-name-is-alread 
or https://stackoverflow.com/questions/64580656/installing-plugin-fatal-error-cannot-declare-class-stripe-stripe-because-the

Now, a possible fix (untested), could be to replace in the file eme-payments.php the line
`require_once 'payment_gateways/stripe/init.php';
`
with
```
if  (!class_exists('\Stripe\Stripe')) 
   require_once 'payment_gateways/stripe/init.php';
```
Now the problem is that then I don't have control over the version being used ... 
But first try it (should work), the replacement should be done 5 times in eme-payments.php.

**Comment by dietrichmd** [comment]
Well, that didn't fix it, but not because there was a problem with the code, but because the client had the API keys in backwards.  Go figure.  Thanks @liedekef

**Reply by liedekef** [reply]
So, does this mean the code helped to resolve the first problem and then there was another due to wrong settings?
Because I can include the mentioned code by default in the next version then.
